### PR TITLE
feat(action-bar): Add configuration option for maximum visible buttons

### PR DESCRIPTION
**Pull request by vibdev**
## Summary

Fixes: #2155 

This allows for more flexible designs using the action bar. The maximum number of visible buttons is now configurable allowing anywhere from 1 -> items.length to be show if there is space. I didn't see any tests for action bar so I added some visual testing rows. Let me know if you want anything else.

## Release Category
Components

---

## Checklist

- [x] MDX documentation adheres to Canvas Kit's [standard MDX template](https://github.com/Workday/canvas-kit/discussions/1131)
- [x] Label `ready for review` has been added to PR

## For the Reviewer

- [ ] PR title is short and descriptive
- [ ] PR summary describes the change (Fixes/Resolves linked correctly)

## Screenshots or GIFs (if applicable)

Screenshot 2023-04-23 at 12 09 04 PM

## Thank You Gif

a man shouting 'i need more' in front of a chalkboard filled with inscrutable formulas

### CODE REVIEW

**Review by RayRedGoose — COMMENTED** [review]
Thanks for adding this, Rusty! 
I've added small suggestions to visual test props. We also need to add an example to [documentation](https://github.com/Workday/canvas-kit/blob/master/modules/react/action-bar/stories/ActionBar.stories.mdx) with explanation of this prop and cypress test for the new story.

Let me know if you need any help with it or want me to add it.

**Comment by vibdev** [comment]
@RayRedGoose Added some docs. I also added React-Test lib tests instead of cypress. There was already a file for that and I didn't see any existing cypress tests. Let me know if that works for y'all.

File: modules/react/action-bar/spec/ActionBar.spec.tsx [hunk 1]
```diff
@@ -1,3 +1,5 @@
+import React from 'react';
+import {getAllByRole, render} from '@testing-library/react';
```
Comment (RayRedGoose) [inline modules/react/action-bar/spec/ActionBar.spec.tsx:2]:
```suggestion
import React from 'react';
import {getAllByRole, render} from '@testing-library/react';
```

**Review by RayRedGoose — COMMENTED** [review]
Code looks good to me! I've just added cypress test for Action Bar (I forgot that we didn't have cypress ActionBar testing file in v8 but test improvement is done for v9). I also did small changes such as removing unused imports or changing relative import to package.

I need to double check the base branch for this PR as we are going to release v9 in 2 weeks. If everything is okay this PR can be merged today.

**Comment by vibdev** [comment]
@RayRedGoose thank you! I am afk for the next week. But appreciate the quick help ❤️

File: modules/react/action-bar/stories/visual-testing/stories_action-bar.tsx [hunk 2]
```diff
@@ -57,23 +56,47 @@ export const ActionBarWithOverflowMenuStates = () => {
...
+          },
+          {
+            label: 'Maximum Visible Items (280px width)',
+            props: {maximumVisible: items.length, containerWidth: 280},
+          },
+        ]}
+        columnProps={[
+          {label: 'Example', props: {}},
+          // {label: '440px container width', props: {containerWidth: '440px'}},
+          // {label: '320px container width', props: {containerWidth: '320px'}},
```
Comment (RayRedGoose) [inline modules/react/action-bar/stories/visual-testing/stories_action-bar.tsx:93]:
```suggestion
```

**Pull request merged**